### PR TITLE
[cli] Preparing to deprecate wrapping of external commands

### PR DESCRIPTION
**Pull request by john-bodley**
This PR indicates to users that the Superset CLI commands which wrap external commands such as `gunicorn`, `celery`, and `flower` are deprecated prior to removing the functionality at a later stage. 

The reason for the change is:
1. All these commands can be run via the delegated application.
2. The current approach only exposes a limited number of command line arguments.
3. It provides unnecessary dependencies which aren't core to the application.

to: @fabianmenges @mistercrunch

### CODE REVIEW

**Comment by xrmx** [comment]
This looks like painful for newcomers coming from outside the python world who have no clue on how to deploy a wsgi app though. The fact that superset looked like an http service while it's just a flask application it's handy to me, also quite "cloud native" :) Would it be possible to keep the dependencies and just remove the wrappers? Anyway if we remove the wrapper we have to provide the equivalent of superset runserver in the docs if we don't want to get swamped with github issues please.

**Comment by mistercrunch** [comment]
Having some docker / docker-compose predefined setup(s) could help compensate in a better-than-its-now kind of way.

Personally cannot commit time to this but I think that'd be the modern alternative for good and easy setup.

File: docs/installation.rst [hunk 1]
```diff
@@ -521,6 +513,10 @@ have the same configuration.
 
     CELERY_CONFIG = CeleryConfig
 
+To start a Celery worker to leverage the configuration run: ::
+
+    celery worker --app=superset.sql_lab:celery_app -Ofair
```
Comment (fabianmenges) [inline docs/installation.rst]: For celery the gevent flag is `-P gevent`

File: docs/installation.rst [hunk 2]
```diff
@@ -147,12 +144,8 @@ Gunicorn, preferably in **async mode**, which allows for impressive
...
-application in a way that works well in your environment.
-
-While the `superset runserver` command act as an quick wrapper
-around `gunicorn`, it doesn't expose all the options you may need,
-so you'll want to craft your own `gunicorn` command in your production
-environment. Here's an **async** setup known to work well: ::
+application in a way that works well in your environment. Here's an **async**
+setup known to work well in production: ::
 
  　gunicorn \
```
Comment (fabianmenges) [inline docs/installation.rst]: We are running gunicorn with gevent which make superset more efficient handling requests `-k gevent '

**Review by fabianmenges — APPROVED** [review]

File: docs/installation.rst (hunk 2 above)
Comment (john-bodley) [reply docs/installation.rst]: @fabianmenges the `-k gevent` argument is specified.

File: docs/installation.rst (hunk 1 above)
Comment (john-bodley) [reply docs/installation.rst]: Thanks. I added `--pool=gevent`. I've used the long names as they're more descriptive.

**Comment by john-bodley** [comment]
@fabianmenges are you ok with the changes I made per your request?

**Review by fabianmenges — APPROVED** [review]

**Review by jeffreythewang — COMMENTED** [review]
I do somewhat agree with @xrmx that it may be painful for newcomers, at least for the `gunicorn` wrapper. I still (and will probably still) use the superset cli wrapper as a shortcut for quickly spinning up an app locally.

File: superset/config.py [hunk 3]
```diff
@@ -42,8 +42,8 @@
 VIZ_ROW_LIMIT = 10000
 # max rows retrieved by filter select auto complete
 FILTER_SELECT_ROW_LIMIT = 10000
-SUPERSET_WORKERS = 2
-SUPERSET_CELERY_WORKERS = 32
+SUPERSET_WORKERS = 2  # deprecated
+SUPERSET_CELERY_WORKERS = 32  # deprecated
 
 SUPERSET_WEBSERVER_ADDRESS = '0.0.0.0'
 SUPERSET_WEBSERVER_PORT = 8088
```
[the remark this answers is not in the document]
Comment (john-bodley) [reply superset/config.py]: @jeffreythewang am I missing other config variables from the commands which are slated to be deprecated?

[the remark this answers is not in the document]
Comment (jeffreythewang) [reply superset/config.py]: Sorry for the late response. I primarily meant it as adding a `# deprecated` comment, both in `cli.py` and the sample `config.py`, to the environment variables that are deprecated. I might be wrong, but I only see `SUPERSET_WEBSERVER_PORT`, `SUPERSET_WEBSERVER_ADDRESS`, and `FLASK_USE_RELOAD` used in `cli.py`.

**Pull request merged**